L1Trigger/L1TMuonCPPF/src/RecHitProcessor.cc

```diff
@@ -7,504 +14,489 @@ RecHitProcessor::~RecHitProcessor() {}
-           rechit_it++) {
-        // const RPCDetId& rpcId = rechit_it->rpcId();
-        int rawId = rpcId.rawId();
-        // int station = rpcId.station();
-        int Bx = rechit_it->BunchX();
-        int isValid = rechit_it->isValid();
-        int firststrip = rechit_it->firstClusterStrip();
-        int clustersize = rechit_it->clusterSize();
-        LocalPoint lPos = rechit_it->localPosition();
-        const RPCRoll *roll = rpcGeom->roll(rpcId);
-        const BoundPlane &rollSurface = roll->surface();
-        GlobalPoint gPos = rollSurface.toGlobal(lPos);
-        float global_theta = emtf::rad_to_deg(gPos.theta().value());
-        float global_phi = emtf::rad_to_deg(gPos.phi().value());
-        //::::::::::::::::::::::::::::
-        // Establish the average position of the rechit
-        int rechitstrip = firststrip;
+  for (const auto &&rpcdgIt : (*rpcDigis)) {
+    const RPCDetId &rpcId = rpcdgIt.first;
+    const RPCDigiCollection::Range &range = rpcdgIt.second;
+    if (rpcId.region() == 0)
+      continue;
 
-        if (clustersize > 2) {
-          int medium = 0;
-          if (clustersize % 2 == 0)
-            medium = 0.5 * (clustersize);
-          else
-            medium = 0.5 * (clustersize - 1);
-          rechitstrip += medium;
+    CPPFClusterizer clizer;
+    CPPFClusterContainer tcls = clizer.doAction(range);
+    CPPFMaskReClusterizer mrclizer;
+    CPPFRollMask mask;
+    CPPFClusterContainer cls = mrclizer.doAction(rpcId, tcls, mask);
+
+    for (auto cl : cls) {
+      int region = rpcId.region();
+      int isValid = rpcDigis.isValid();
+      //      int rawId = rpcId.rawId();
+      int Bx = cl.bx();
+      const int firststrip = cl.firstStrip();
+      const int clustersize = cl.clusterSize();
+      const int laststrip = cl.lastStrip();
+      const RPCRoll *roll = rpcGeom->roll(rpcId);
+      // Get Average Strip position
+      const float fstrip = (roll->centreOfStrip(firststrip)).x();
+      const float lstrip = (roll->centreOfStrip(laststrip)).x();
+      const float centreOfCluster = (fstrip + lstrip) / 2;
+      const double y = cl.hasY() ? cl.y() : 0;
+      LocalPoint lPos(centreOfCluster, y, 0);
+      if (roll->id().region() != 0) {
+        const auto &topo = dynamic_cast<const TrapezoidalStripTopology &>(roll->topology());
+        const double angle = topo.stripAngle((firststrip + laststrip) / 2.);
+        const double x = centreOfCluster - y * std::tan(angle);
+        lPos = LocalPoint(x, y, 0);
+      }
+      const BoundPlane &rollSurface = roll->surface();
+      GlobalPoint gPos = rollSurface.toGlobal(lPos);
+      float global_theta = emtf::rad_to_deg(gPos.theta().value());
+      float global_phi = emtf::rad_to_deg(gPos.phi().value());
+
+      // Endcap region only
+      if (region != 0) {
+        int int_theta =
+            (region == -1 ? 180. * 32. / 36.5 : 0.) + (float)region * global_theta * 32. / 36.5 - 8.5 * 32 / 36.5;
```

Comment:
For the future, please define constants instead of using "magic numbers" 
7.6 of https://cms-sw.github.io/cms_coding_rules.html
(I see these numbers were already existing in the file)